### PR TITLE
Add dashboard tab

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr uses the changes from #200 to implement a first dashboard page with active and favorite decks.

### CODE REVIEW

File: frontend/types/Deck.ts [hunk 1]
```diff
@@ -5,8 +5,8 @@ export type Deck = {
 	deckID: string;
 	deckName: string;
 	deckType: "PUBLIC" | "PRIVATE";
-	isActive?: boolean;
-	isFavorite?: boolean;
+	active?: boolean;
+	favorite?: boolean;
```
Comment (TomRomeo) [inline frontend/types/Deck.ts:9]: Please rename to `isActive` and `isFavorite` again

**Review by TomRomeo — CHANGES_REQUESTED** [review]

**Review by memeToasty — APPROVED** [review]

**Review by TomRomeo — APPROVED** [review]

File: frontend/components/deck/Deck.tsx [hunk 2]
```diff
@@ -60,10 +61,14 @@ export const Deck = ({
...
 }: DeckProps) => {
 	const router = useRouter();
-	const { _ } = useLingui();
+	const { mutate } = useSWRConfig();
 
 	const [isFavorite, setFavorite] = useState(deck.isFavorite);
 
+	useEffect(() => {
+		setFavorite(deck.isFavorite);
+	});
```
Comment (4KevR) [inline frontend/components/deck/Deck.tsx:70]: Dependencies of `useEffect` are not declared

File: frontend/components/navigation/Tabs/DashboardTab.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export const DashboardTab = ({ className = "" }: DashboardTabProps) => {
+	const { decks: activeDecks } = useActiveDecks();
+	const { decks: favoriteDecks } = useFavoriteDecks();
+
+	const { _ } = useLingui();
+
+	return (
+		<div className={`space-y-5 ${className}`}>
+			<DeckList header={_(msg`Active decks`)} decks={activeDecks} />
+			<DeckList header={_(msg`Favorite Decks`)} decks={favoriteDecks} />
```
Comment (4KevR) [inline frontend/components/navigation/Tabs/DashboardTab.tsx:26]: Either both decks capitalized or none of them

**Review by 4KevR — CHANGES_REQUESTED** [review]
Great job, see comments for small nits

File: frontend/components/deck/Deck.tsx (hunk 2 above)
Comment (Julian702) [reply frontend/components/deck/Deck.tsx]: fixed in 7c65439370d48f47cd0c3f4e2886b9a8ebff51d1

File: frontend/components/navigation/Tabs/DashboardTab.tsx (hunk 3 above)
Comment (Julian702) [reply frontend/components/navigation/Tabs/DashboardTab.tsx]: fixed in 7c65439370d48f47cd0c3f4e2886b9a8ebff51d1

**Review by 4KevR — APPROVED** [review]

**Pull request merged**